utilities/version_utils.py

```
@@ -0,0 +1,89 @@
+
+def compare_versions(version1: str, version2: str) -> int:
+    """
+    Compare two version strings.
+    Returns: -1 if version1 < version2, 0 if equal, 1 if version1 > version2
+    """
+    from packaging import version
+
+    try:
+        return (version.parse(version1) > version.parse(version2)) - (version.parse(version1) < version.parse(version2))
+    except Exception as e:
+        LOGGER.error(f"[VERSION COMPARE] Error comparing versions: {e}")
```

Ruff (0.11.9)

Undefined name `LOGGER`

(F821)